addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -377,7 +377,7 @@ const Wysiwyg = Widget.extend({
                         res_model: resModel,
                         res_id: parseInt(resId),
                         data: (isBackground ? el.dataset.bgSrc : el.getAttribute('src')).split(',')[1],
-                        mimetype: el.dataset.mimetype,
```

Comment:
I find this fix really strange, the same way I found Arthur's implementation of mimetype in the `cleanForSave` strange too. For me, the data-mimetype should directly be set to svg by the option which actually add the shape/convert to SVG. And the original mimetype be mentioned directly too and used where it is actually needed.

Reply:
I Tested both cases and it seems that code was implemented this way to set the right mimetype if image changed...
Let's suppose that we have a JPG image and we moved the code updating data-mimetype to option's method :

- After applying a shape on it, if we replace the image by a PNG image, we'll get as a result an image with
dataset = {..., mimetype: 'image/png', originalMimetype: 'image/jpeg'} (an update to SVG is always needed)

I think we can keep it this way or add it in options method in a way to keep the real mimetype when image changed by user. [Added a separate commit for the second implementation(*)].

(*) : I added commits by "push" as requested in task PAD with a "v-i" to track changes of a single commit.

Reply:
I am not sure to understand everything that you are saying here. My point was to adapt all related code so that any option changing the image should directly lead to a consistent `mimetype` + `originalMimetype`. So in your example: ` {..., mimetype: 'image/png', originalMimetype: 'image/jpeg'}` should simply not be possible -> you change the image, both values are changed, you add a shape, only mimetype is changed, etc etc.

Did not look at your code update yet, is this what you did?

Reply:
Actually, it's the same idea, the `mimetype`/`orignalMimetype` keep the right values when image changed & option applied.
I added an update on the code to handle the case where shape removed (`data-mimetype` should get the original
value and `data-original-mimetype` should be removed).